OCPBUGS-8097-re: Updated ELB nomenclature throughout docs

dfitzmau opened the pull request:
[OCPBUGS-8097](https://issues.redhat.com/browse/OCPBUGS-8097)

Version(s):
4.14 through to 4.10

Link to docs preview:
* [AWS account limits-OCD](https://61919--docspreview.netlify.app/openshift-dedicated/latest/osd_planning/aws-ccs.html#aws-limits_aws-ccs)
* [Provisioned AWS Infrastructure-OCD](https://61919--docspreview.netlify.app/openshift-dedicated/latest/osd_planning/aws-ccs.html#ccs-aws-provisioned_aws-ccs)
* [AWS account limits](https://61919--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_aws/installing-aws-account.html#installation-aws-limits_installing-aws-account)
* [Provisioned AWS Infrastructure-ROSA](https://61919--docspreview.netlify.app/openshift-rosa/latest/rosa_planning/rosa-sts-aws-prereqs#rosa-aws-policy-provisioned_rosa-sts-aws-prereqs)


QE review:
- [x] QE has approved this change.
<!--- QE approval is required to merge a PR except for changes that do not impact the meaning of the docs. --->

Additional information:
* [OSDOCS-5440](https://github.com/openshift/openshift-docs/pull/55321)

dfitzmau commented on modules/dedicated-configuring-your-application-routes.adoc:
```diff
@@ -5,7 +5,7 @@
 = Configuring your application routes
 
-When your cluster is provisioned, an AWS elastic load balancer (ELB) is created
+When your cluster is provisioned, an AWS elastic load balancing (ELB) load balancer is created
```
In OCP Dedicated 3 docs but not 4.

codymant replied:
@dfitzmau AWS does not use the term "AWS elastic load balancing". The full service name is always "Elastic Load Balancing" by itself, without "AWS". The name should be capitalized also. 

Suggested revision: "When your cluster is provisioned, an Elastic Load Balancing  (ELB) load balancer is created.

codymant commented on modules/aws-limits.adoc:
```diff
@@ -50,7 +50,7 @@ To use the `us-east-1` region, you must increase the EIP limit for your account.
 |3
 |20 per region
-|By default, each cluster creates internal and external network load balancers for the primary API server and a single classic elastic load balancer for the router. Deploying more Kubernetes LoadBalancer Service objects will create additional link:https://aws.amazon.com/elasticloadbalancing/[load balancers].
+|By default, each cluster creates internal and external network load balancers for the primary API server and a single classic ELB load balancer for the router. Deploying more Kubernetes LoadBalancer Service objects will create additional link:https://aws.amazon.com/elasticloadbalancing/[load balancers].
```
@dfitzmau The following comments are derived from AWS style guidance. 

On first usage of an AWS service name within a topic, AWS style recommends writing out the full name to introduce the acronym (e.g "Elastic Load Balancing (ELB)"). The acronym may be used on subsequent usage, once the full service name is introduced. 

When referring to specific load balancer types, AWS uses the following terminology. These words are always used together in the phrasing seen below, and written in all caps. 

- Classic Load Balancer 
- Application Load Balancer 
- Network Load Balancer 

See: 

https://docs.aws.amazon.com/elasticloadbalancing/latest/classic/introduction.html
https://docs.aws.amazon.com/elasticloadbalancing/latest/application/introduction.html
https://docs.aws.amazon.com/elasticloadbalancing/latest/network/introduction.html

codymant replied:
When referring to a generic load balancer and type does not matter (i.e. Classic, Application, or Network), the term "load balancer" is used.

codymant replied:
I recommend revising as follows: 

"By default, each cluster creates internal and external ELB Network Load Balancers for the primary API server and a single Classic Load Balancer for the router."

dfitzmau replied:
Thanks, @codymant . I updated the sentence based on your feedback. I updated the component name, `ELB/NLB`, to `Elastic Load Balancing (ELB)`. I removed `NLB` as it seemed to confuse things. WDYT?

codymant commented on modules/aws-limits.adoc:
```diff
@@ -63,7 +63,7 @@ To use the `us-east-1` region, you must increase the EIP limit for your account.
 |The default installation creates 21 ENIs and an ENI for each availability zone in your region. For example, the `us-east-1` region contains six availability zones, so a cluster that is deployed in that zone uses 27 ENIs. Review the link:https://aws.amazon.com/about-aws/global-infrastructure/[AWS region map] to determine how many availability zones are in each region.
 
-Additional ENIs are created for additional machines and elastic load balancers that are created by cluster usage and deployed workloads.
+Additional ENIs are created for additional machines and ELB load balancers that are created by cluster usage and deployed workloads.
```
@dfitzmau I would consider dropping the "ELB" here. Earlier in the topic, it's already been introduced that this information applies to ELB resources. The acronym seems unnecessary.

codymant commented on modules/installation-aws-limits.adoc:
```diff
@@ -69,7 +69,7 @@ To use the `us-east-1` region, you must increase the EIP limit for your account.
 |20 per region
 |By default, each cluster creates internal and external network load balancers for the master
-API server and a single classic elastic load balancer for the router. Deploying
+API server and a single classic ELB load balancer for the router. Deploying
```
@dfitzmau The heading above intros the service name, so there's no need to repeat here. Suggested revision: "single Classic Load Balancer for the router."

codymant commented on modules/rosa-aws-provisioned.adoc:
```diff
@@ -87,7 +87,7 @@ image::VPC-Diagram.png[VPC Reference Architecture]
 == Security groups
 
-AWS security groups provide security at the protocol and port access level; they are associated with EC2 instances and Elastic Load Balancers. Each security group contains a set of rules that filter traffic coming in and out of an EC2 instance. You must ensure the ports required for the OpenShift installation are open on your network and configured to allow access between hosts.
+AWS security groups provide security at the protocol and port access level; they are associated with EC2 instances and Elastic Load Balancing (ELB) load balancers. Each security group contains a set of rules that filter traffic coming in and out of an EC2 instance. You must ensure the ports required for the OpenShift installation are open on your network and configured to allow access between hos ...
```
@dfitzmau Should say "Amazon EC2 instance(s)", instead of "EC2 instance".

yunjiang29 commented:
@lihongan can you take care of this PR?

lihongan commented on modules/aws-limits.adoc:
```diff
@@ -47,10 +47,10 @@ To use the `us-east-1` region, you must increase the EIP limit for your account.
 |3
 |20 per region
-|By default, each cluster creates internal and external network load balancers for the primary API server and a single classic elastic load balancer for the router. Deploying more Kubernetes LoadBalancer Service objects will create additional link:https://aws.amazon.com/elasticloadbalancing/[load balancers].
+|By default, each cluster creates internal and external ELB Network Load Balancers for the primary API server and a single Classic Load Balancer for the router. Deploying more Kubernetes LoadBalancer Service objects will create additional link:https://aws.amazon.com/elasticloadbalancing/[load balancers].
```
I think `each cluster creates internal and external Network Load Balancers for the primary API server ...` is OK, no need to mention ELB here.

dfitzmau replied:
Thanks. Updated.

lihongan commented on modules/ccs-aws-provisioned.adoc:
```diff
@@ -46,7 +46,7 @@ Volume requirements for each EC2 instance:
 
 [id="aws-policy-elastic-load-balancers_{context}"]
-== Elastic load balancers
+== Elastic load balancing (ELB) load balancers
```
`Elastic Load Balancing (ELB) load balancers` is better, please capitalize L and B

dfitzmau replied:
Well spotted. Updated.

lihongan commented:
other looks good. thank you!

dfitzmau commented:
/label peer-review-needed

skopacz1 commented:
/label peer-review-in-progress

adellape reviewed (commented):
@skopacz1 Peer review comments LGTM 👍 

@dfitzmau Also please note in comment 0 which versions to cherry-pick this to.

dfitzmau commented on modules/rosa-aws-provisioned.adoc:
```diff
@@ -87,7 +87,7 @@ image::VPC-Diagram.png[VPC Reference Architecture]
 == Security groups
 
-AWS security groups provide security at the protocol and port access level; they are associated with EC2 instances and Elastic Load Balancers. Each security group contains a set of rules that filter traffic coming in and out of an EC2 instance. You must ensure the ports required for the OpenShift installation are open on your network and configured to allow access between hosts.
+AWS security groups provide security at the protocol and port access level; they are associated with EC2 instances and Elastic Load Balancing (ELB) load balancers. Each security group contains a set of rules that filter traffic coming in and out of an EC2 instance or EC2 instances. You must ensure the ports required for the OpenShift installation are open on your network and configured to allow a ...
```
Thanks. Option B sounds good to me.

dfitzmau commented on modules/installation-aws-limits.adoc:
I checked this file and do not see the "ELB/NLB" reference you discuss. Are you viewing the latest commit?

dfitzmau commented on modules/aws-limits.adoc:
Good catch. Updated.

dfitzmau commented:
Hi @skopacz1 and @adellape . Thank you for the reviews. I updated the PR based on your feedback. I reached out to Cody (an AWS writer) and I dropped the BLN and CLB abbreviations as they are not official abbreviations. See/read:

- https://docs.aws.amazon.com/elasticloadbalancing/latest/network/introduction.html
- https://docs.aws.amazon.com/elasticloadbalancing/latest/classic/introduction.html

dfitzmau commented:
/label merge-review-needed

jeana-redhat reviewed (approved):
LGTM

/remove-label merge-review-in-progress
/remove-label merge-review-needed

jeana-redhat commented:
/cherrypick enterprise-4.14

jeana-redhat commented:
/cherrypick enterprise-4.13

jeana-redhat commented:
/cherrypick enterprise-4.12

jeana-redhat commented:
/cherrypick enterprise-4.11

jeana-redhat commented:
/cherrypick enterprise-4.10

jeana-redhat commented:
@dfitzmau 4.10&4.11 need a manual CP, sorry :bow:
